readthedocs/starterguides/developerguides/Orchestration_Developer_Guide.md

```diff
@@ -0,0 +1,367 @@
+  snapshot_id:
+    type: string
+    required: false
+  snapshot_from_cloud:
+    type: string
+    required: false
+  auth_token:
+    type: string
+    required: false
+runner_type: "python-script"
+```
+### Points to remember:
+    a) For every action/task, there should be an entry point. entry_point is the script to execute this action. This can either be a python script or a shell script. OpenSDS prefers python script for using the rich python libraries, as required
+    b) All the parameters of this actions are detailed here. It should define the 'type' of the parameter, whether this parameter is 'required' or not and if the pramaeter can have some 'default' value
+    c) The action should specify 'runner_type'. Ex: runner_type: "python_script"
+
+Example of python runner script (create_volume.py):
+```python
+import requests
+import json
+
+from st2common.runners.base_action import Action
+
+
+class CreateVolumeAction(Action):
+    def run(self,
+            ip_addr="",
+            port="",
+            tenant_id="",
+            name="",
+            description="Volume",
+            availability_zone="default",
+            profile_id="",
+            snapshot_id="",
+            snapshot_from_cloud="",
+            auth_token="",
+            size=1):
+        data = {
+            "Name": name,
+            "Description": description,
+            "AvailabilityZone": availability_zone,
+            "ProfileId": profile_id,
+            "SnapshotId": snapshot_id,
+            "SnapshotFromCloud": snapshot_from_cloud,
+            "Size": size
+            }
+        if profile_id:
+            data["ProfileId"] = profile_id
+
+        url = "http://" + \
+            ip_addr + ":" + \
+            port + "/v1beta/" + \
+            tenant_id + "/block/volumes"
+
+        headers = {
+            'content-type': 'application/json',
+            'x-auth-token': auth_token
+        }
+        r = requests.post(url=url, data=json.dumps(data), headers=headers)
+        r.raise_for_status()
+        resp = r.json()
+        return resp["id"]
+```
+### Points to remember:
+    a) Every python script need to import 'Action' from st2common.eunners.base_action
+    b) Define your action class which inherits from 'Action' base class
+    c) This class must define the 'run' function
+    d) All the business logic to execute the action goes into this 'run' method
+    e) Parameters for the 'run' method is the class object ('self' here) and all the required parameters for this action. These parameters are same that was defined in the corresponding action YAML
+
+## How to load it into the Orchestration Manager
+Once the workflows, actions and runners are defined, these workflows need to be registered into the Orchestration Manager, StackStorm.
+If the StackStorm is installed using Docker image:
+Go to the interactive bash cell of the StackStorm docker instance and follow the steps below
+Else
+Go to the command prompt and execute:
```

Comment:
Please combine lines 199-202 into one paragraph.  The newlines won't have any effect anyway in the resulting doc.